src/platform/Darwin/BleConnectionDelegateImpl.mm

```diff
@@ -59,25 +62,28 @@ @interface MTRBleConnection : NSObject <CBCentralManagerDelegate, CBPeripheralDe
 
 @property (strong, nonatomic) dispatch_queue_t workQueue; // the CHIP work queue
 @property (strong, nonatomic) CBCentralManager * centralManager;
-@property (strong, nonatomic) CBPeripheral * peripheral;
+@property (strong, nonatomic, nullable) CBPeripheral * peripheral;
+// The discriminator that matches "peripheral"
+@property (strong, nonatomic, nullable) NSNumber * matchedLongDiscriminator;
 @property (strong, nonatomic, readonly, nullable) dispatch_source_t timer;
 @property (assign, nonatomic, readonly) BleConnectionMode currentMode;
 @property (strong, nonatomic) NSMutableDictionary<CBPeripheral *, NSDictionary *> * cachedPeripherals;
 @property (assign, nonatomic) bool found;
-@property (assign, nonatomic) chip::SetupDiscriminator deviceDiscriminator;
+@property (assign, nonatomic) std::vector<chip::SetupDiscriminator> desiredDiscriminators;
```

Comment:
I guess this fits with the existing style, but we should ideally stop declaring things as properties where only the ivars are needed, and just declare the ivar itself.

Reply:
Yeah, I just matched style for now, as you said...